apollo-router/src/services/subgraph_service.rs

```diff
@@ -914,6 +1058,76 @@ fn get_websocket_request(
     Ok(request)
 }
 
+fn get_sse_client(
+    service_name: String,
+    parts: http::request::Parts,
+    subgraph_sse_cfg: &SseConfiguration,
+) -> Result<crate::protocols::sse::client::ClientBuilder, FetchError> {
+    let subgraph_url = url::Url::parse(&parts.uri.to_string()).map_err(|err| {
+        tracing::error!("cannot parse subgraph url {}: {err:?}", parts.uri);
+        FetchError::SubrequestHttpError {
+            service: service_name.clone(),
+            reason: "cannot parse subgraph url".to_string(),
+            status_code: None,
+        }
+    })?;
+
+    let subgraph_url = match &subgraph_sse_cfg.path {
+        Some(path) => subgraph_url
+            .join(path)
+            .map_err(|_| FetchError::SubrequestHttpError {
+                service: service_name.clone(),
+                reason: "cannot parse subgraph url with the specific sse path".to_string(),
+                status_code: None,
+            })?,
+        None => subgraph_url,
+    };
+
+    let mut builder = ClientBuilder::for_url(subgraph_url.as_str()).map_err(|e| {
+        FetchError::SubrequestHttpError {
+            service: service_name.clone(),
+            reason: format!("cannot parse subgraph url with the specific sse path {e:?}"),
+            status_code: None,
+        }
+    })?;
+
+    builder = builder.reconnect(
+        ReconnectOptions::reconnect(subgraph_sse_cfg.reconnect.unwrap_or(true))
+            .retry_initial(subgraph_sse_cfg.retry_initial.unwrap_or(true))
+            .delay(
+                subgraph_sse_cfg
+                    .delay
+                    .unwrap_or_else(|| Duration::from_secs(1)),
+            )
+            .backoff_factor(subgraph_sse_cfg.backoff_factor.unwrap_or(2))
+            .delay_max(
+                subgraph_sse_cfg
+                    .delay
+                    .unwrap_or_else(|| Duration::from_secs(60)),
+            )
+            .build(),
+    );
+
+    for (key, value) in parts.headers.iter() {
+        builder = builder
+            .header(key.as_str(), value.to_str().unwrap_or_default())
+            .map_err(|e| FetchError::SubrequestHttpError {
+                service: service_name.clone(),
+                reason: format!("can not add header to sse client: {e:?}"),
+                status_code: None,
+            })?;
+    }
+    builder = builder.header("Accept", "text/event-stream").map_err(|e| {
```

Comment:
Could you use the constant https://docs.rs/http/latest/http/header/constant.ACCEPT.html ?